### PR TITLE
apply_patches functionality

### PR DESCRIPTION
**Pull request by darx0r**
apply patches made in IDAPython on the binary file (custom path can be supplied)

### CODE REVIEW

File: sark/data.py [hunk 1]
```diff
@@ -111,6 +116,30 @@ def collector(ea, fpos, original, patched):
     return patched_bytes
 
 
+def apply_patches(patched_path=None):
+    to_patch = idaapi.get_input_file_path()
```
Comment (tmr232) [inline sark/data.py]: There does not have to be an input file.

File: sark/data.py [hunk 2]
```diff
@@ -111,6 +116,30 @@ def collector(ea, fpos, original, patched):
...
+
+    patches = get_patched_bytes()
+
+    with open(to_patch, "r+b") as output:
+
+        for patch in patches.itervalues():
+
+            output.seek(patch.fpos)
+            curr_byte = ord(output.read(1))
+            if curr_byte != patch.original:
```
Comment (tmr232) [inline sark/data.py]: Will this not fail if we are re-patching the original file?

File: sark/data.py (hunk 1 above)
Comment (darx0r) [inline sark/data.py]: In a case of no input file - indicative IO exception will raised
In a case of additional binary loading - only the base binary (which the IDB was created for) path will be returned from idaapi.get_input_file_path, which is reasonable.

I don't see a problem here

File: sark/data.py (hunk 2 above)
Comment (darx0r) [inline sark/data.py]: It's a decision to make.
Relaxing this validation will make our life easier :)

(seems like there is no API to update the original value of a patched byte)

**Comment by tmr232** [comment]
Please rebase your changes onto the master branch.
Other than that, looks good.

**Comment by darx0r** [comment]
I don't have write access to the repo. 
Please use "Rebase and merge" as demonstrated here: https://github.com/blog/2243-rebase-and-merge-pull-requests

**Comment by tmr232** [comment]
I rebased merged your code manually, seems like github didn't like it. I am closing this PR as the code is now in the project.

Sorry for taking so long.

**Pull request closed without merging**